spec/models/regulator_spec.rb

```diff
@@ -92,4 +106,53 @@
       end
     end
   end
+
+  describe "#batch_inspect!" do
+    context "with a few records" do
+      before do
+        ActiveRecord::Base.transaction do
+          10.times { create(:reading, :day_time, {temp: 75, outdoor_temp: 50 }) }
+        end
+      end
+
+      it "updates violation status in batches", :vcr do
+        expect(Reading.where(violation: true)).to have(0).items
+
+        Reading.order(created_at: :asc).first.update(temp: 65)
+        Reading.order(created_at: :asc).last.update(temp: 65)
+
+        Regulator.new(Reading.all).batch_inspect!(silent: true)
+        expect(Reading.where(violation: true)).to have(2).items
+      end
+    end
+
+    context "with many records" do
```

Comment:
Oh man, this find_in_batches method is perfect! Great suggestion Oliver, I wish I'd known about that sooner. Good point about NewRelic, I've been meaning to get that set up. Several people have pointed out that we need it. I've set it up on projects before, surprised I still haven't done it for this one yet.